packages/did-auth-siop-op-authenticator/src/agent/DidAuthSiopOpAuthenticator.ts

```diff
@@ -0,0 +1,204 @@
+import { registerCustomApprovalForDidSiopArgs, schema } from '../index'
+import { IAgentPlugin } from '@veramo/core'
+import {
+  PassBy,
+  ResponseMode,
+  ParsedAuthenticationRequestURI,
+  PresentationDefinitionWithLocation,
+  VerifiedAuthenticationRequestWithJWT,
+  VerifyAuthenticationRequestOpts,
+  VerifiablePresentationTypeFormat,
+  VerificationMode,
+} from '@sphereon/did-auth-siop/dist/main/types/SIOP.types'
+import { OP, PresentationExchange } from '@sphereon/did-auth-siop/dist/main'
+import { SubmissionRequirementMatch, VerifiableCredential } from '@sphereon/pe-js'
+import {
+  events,
+  IAuthenticateWithDidSiopArgs,
+  IAuthRequestDetails,
+  IDidAuthSiopOpAuthenticator,
+  IDidAuthSiopOpAuthenticatorArgs,
+  getDidSiopAuthenticationRequestDetailsArgs,
+  IGetDidSiopAuthenticationRequestFromRpArgs,
+  IMatchedPresentationDefinition,
+  IRequiredContext,
+  ISendDidSiopAuthenticationResponseArgs,
+  IVerifyDidSiopAuthenticationRequestUriArgs,
+} from '../types/IDidAuthSiopOpAuthenticator'
+
+const fetch = require('cross-fetch')
+
+/**
+ * {@inheritDoc IDidAuthSiopOpAuthenticator}
+ */
+export class DidAuthSiopOpAuthenticator implements IAgentPlugin {
+  readonly schema = schema.IVcApiVerfierAgentPlugin
+  readonly methods: IDidAuthSiopOpAuthenticator = {
+    authenticateWithDidSiop: this.authenticateWithDidSiop.bind(this),
+    getDidSiopAuthenticationRequestFromRP: this.getDidSiopAuthenticationRequestFromRP.bind(this),
+    getDidSiopAuthenticationRequestDetails: this.getDidSiopAuthenticationRequestDetails.bind(this),
+    verifyDidSiopAuthenticationRequestURI: this.verifyDidSiopAuthenticationRequestURI.bind(this),
+    sendDidSiopAuthenticationResponse: this.sendDidSiopAuthenticationResponse.bind(this),
+    registerCustomApprovalForDidSiop: this.registerCustomApprovalForDidSiop.bind(this),
+  }
+  private readonly op: OP
+  private readonly did: string
+  private readonly kid: string
+  private readonly privateKey: string
+  private readonly expiresIn: number | undefined
+  private readonly didMethod: string | undefined
+  private readonly customApprovals: Record<string, (verifiedAuthenticationRequest: VerifiedAuthenticationRequestWithJWT) => Promise<void>>
+
+  constructor(options: IDidAuthSiopOpAuthenticatorArgs) {
+    this.did = options.did
```

Comment:
Hmzz, I would expect Veramo identifiers to be used, instead of DID strings, same for KID and privateKey